src/smc-webapp/r_misc.cjsx

```diff
@@ -185,11 +185,15 @@ exports.Loading = Loading = rclass
     displayName : 'Misc-Loading'
 
     propTypes :
-        style : rtypes.object
+        style  : rtypes.object
+        text   : rtypes.string
+
+    getInitialState : ->
+        text   : @props.text ? 'Loading...'
```

Comment:
well, I don't fully get that, but ok. So, if it is done with `getInitialState`, changing this props.text won't change the text?

Because I've mostly copied over what I saw in the file and when I scrolled down  I saw `getInitialState: ->  number : @props.number` in `NumberInput`. So, the idea here is, to derive the value of an input widget and make it robust against changes from the parent? If so, I think I understand it.